### PR TITLE
Data view cosmetic changes

### PR DESCRIPTION
**Pull request by ash3rz**
This PR will:
* Switch to single selection when clicking on a row unless the user specifically clicks on the checkbox or holds down the Shift key.  This will make clicking on the Details button much easier because you don't have to deselect the previous item first.
* Enable saving custom columns to local storage (didn't realize how easy this was, would've done it sooner otherwise)
* Align the data's dot menu to the right. Aligning the dot menu to the right means the menu won't move while you're adding/removing columns.
* Change the Settings/Gear icon for customizing columns to text unless the view is small. 
Desktop:
image
Mobile:
image
* Keep the paging footer at the bottom and have the table scroll 🎉 If we like these changes, I can apply this to the other pages using the footer
Scrolling in action - shows changing the height and changing the width (nav bar hides after a point and the footer stays put):
Apr-29-2020 14-28-14

### CODE REVIEW

File: src/components/utils/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        return null;
+    }
+}
+
+export function setLocalStorage(key, value) {
+    try {
+        return window.localStorage.setItem(key, value);
+    } catch (error) {
+        return;
+    }
```
Comment (sri2k1us) [inline src/components/utils/localStorage.js:22]: `setItem` doesn't return anything.

File: src/components/data/listing/CustomizeColumns.js [hunk 2]
```diff
@@ -58,19 +75,24 @@ function CustomizeColumns(props) {
...
                 aria-haspopup="true"
                 size="small"
                 onClick={handleColumnSettingClick}
             >
-                <Settings />
-            </IconButton>
+                <Hidden smUp>
+                    <Settings />
+                </Hidden>
+                <Hidden xsDown>{getMessage("customizeColumns")}</Hidden>
```
[the remark this answers is not in the document]
Comment (psarando) [reply src/components/data/listing/CustomizeColumns.js:94]: Maybe an outlined button would work here?

File: src/components/utils/localStorage.js (hunk 1 above)
Comment (ash3rz) [reply src/components/utils/localStorage.js:22]: Oops, good catch

File: src/components/data/listing/CustomizeColumns.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ash3rz) [reply src/components/data/listing/CustomizeColumns.js:94]: Cool, thanks for the suggestions - I think I'll go with the outlined button option.  It stands out more.

**Comment by ash3rz** [comment]
Thanks for the reviews!  I went ahead and applied the styling changes to the other views using tables.  I also removed a commit I pushed to test out the Chromatic stuff, but left in the commit that fixes the UploadTrackingContext bug.

**Pull request merged**